### ISSUE TITLE
Eviction and NONE policy

### ISSUE THREAD

**Issue by yarmol** [hazelcast/hazelcast#9274]
I am using version 3.3.3.
I have such configuration: 

> 
    <map name="GROUP.CORE.SESSION_REGISTRY.*">
      <time-to-live-seconds>0</time-to-live-seconds>
      <max-idle-seconds>0</max-idle-seconds>
      <eviction-policy>NONE</eviction-policy> </map>

and one more:

>    
      <map name="USER.*">
      <time-to-live-seconds>0</time-to-live-seconds>
      <max-idle-seconds>0</max-idle-seconds>
      <eviction-policy>NONE</eviction-policy>
     </map>

I do not need eviction, because application have full control under all maps and entries. And all unused/old entries will be deleted by application.
But I detect that some entries disapeared. Setting listener on problem maps I get such row in log:.

> 2016-11-18T18:10:15,166  | EntryEvent{entryEventType=EVICTED, member=Member [10.0.2.15]:5701, name='GROUP.CORE.SESSION_REGISTRY.SESSION_TO_USER', key=af45263f-0333-498c-a2ee-f209fe0f2474, oldValue=test-admin@testdev.org, value=null}

And I have a log of such rows in log about eviction.

Is it wildcards are not applicable, or config is wrong? Why hazelcast instance is using some eviction policy? Size of these maps are not big, less than 100.
Do I need to set some additional options to prevent eviction.

**Comment by ahmetmircik** [comment]
Hi @yarmol, 

I did a simple test with an identical configuration, seems there is no eviction happened. In your case, may there be puts with ttl like `map.put(i, i, 1, TimeUnit.SECONDS )`? Also can you send a simple test to show what you did see?

```
public class EvictionTest {

    public static void main(String[] args) throws InterruptedException {
        Config config = new Config();
        MapConfig mapConfig = config.getMapConfig("GROUP.CORE.SESSION_REGISTRY.*");
        mapConfig.setEvictionPolicy(NONE);
        mapConfig.setMaxIdleSeconds(0);
        mapConfig.setTimeToLiveSeconds(0);

        HazelcastInstance hz = Hazelcast.newHazelcastInstance(config);
        IMap map = hz.getMap("GROUP.CORE.SESSION_REGISTRY.SESSION_TO_USER");
        map.addEntryListener(new EntryAdapter() {
            @Override
            public void onEntryEvent(EntryEvent event) {
                System.out.println(event);
            }
        }, true);

        for (int i = 0; i < 1000; i++) {
            map.put(i, i);
        }

        // wait events
        TimeUnit.SECONDS.sleep(10);


    }
}
```

**Comment by jerrinot** [comment]
@yarmol: do you have any update?

**Comment by yarmol** [comment]
Hi, thx a lot for your comments. 
The problem was solved, and config with explict settings of eviction policy as NONE is workable. 
So about the essence of the problem.
I have 2 workable nodes of hazelcast. At first, I have no eviction settings, and system was configured by default (but default is also NONE, and it is strange). After the first "strange eviction event" appeared, I set config as I had wrote in the first post. This new configs was deployed on my 2 nodes. 
But the "events" did not terminate. So I decide to configure entry listeners on problem IMaps. The result of this listeners, you know, the chunk of log with "entryEventType=EVICTED" hint.
But there is another thing, ip - 10.0.2.15 is not belong to my any of my nodes. My nodes,with explict policy settings work correctly. But strange 10.0.2.15 node had default config. It was developer local node. 
The fact is that all hazelcast we have, configured with the same group name 'hazelcast', but we turn off multicast, set explict interface ip, and explict members, and "hazelcast.socket.bind.any" property is also false. We thing that group name is not important, because of these explict LAN settings.
So, after the change of group name everything become fine. No eviction any more.
But I do not understand why does it happend with default configs? And why another 2.15 machine with member/interface settings connected to the neighbor cluster.
My node-1 had:
`
<group>
        <name>hazelcast</name>
 </group>
 <network>
 <public-address>10.1.60.111</public-address>
 <join>
            <multicast enabled="false">
            </multicast>
            <tcp-ip enabled="true">
			    <member>10.1.60.111,10.1.60.112</member>
                <interface>10.1.60.111</interface>
            </tcp-ip>
  </join>
  </network>
`
node-2:
`
<group>
        <name>hazelcast</name>
 </group>
 <network>
 <public-address>10.1.60.112</public-address>
 <join>
            <multicast enabled="false">
            </multicast>
            <tcp-ip enabled="true">
			    <member>10.1.60.111,10.1.60.112</member>
                <interface>10.1.60.112</interface>
            </tcp-ip>
  </join>
  </network>
`
node-2.15
`
<group>
        <name>hazelcast</name>
 </group>
 <network>
 <public-address>10.0.2.15</public-address>
 <join>
            <multicast enabled="false">
            </multicast>
            <tcp-ip enabled="true">
			    <member>10.0.2.15</member>
                <interface>10.0.2.15</interface>
            </tcp-ip>
  </join>
  </network>
`
And bind-any property = false also.
